File: 1874.py
def main():
  #num = int(input())
  #ls = []
  #for i in range(num):
  #  ls.append(int(input()))

  num = 100000
  ls = list(range(100000,0,-1))
  print(ls)
  numbers = ls.copy()
  numbers.sort()

  pos = 0
  stack = []
  act = ''
  stack.append(numbers[0])
  numbers = numbers[1:]
  act = act+'+\n'
  while pos < num:
    while len(stack) == 0 or stack[-1] != ls[pos]:
      if len(numbers) == 0:
        return 'NONE'
      stack.append(numbers[0])
      numbers = numbers[1:]
      act = act+'+\n'
    stack.pop()
    act = act+'-\n'
    pos += 1
  return act

act = main()
print(act)

File: 1874.cc
#include <cstdio>
#include <iostream>
#include <functional>
#include <algorithm>
#include <stack>
#include <string>
#define MAX 100000
using namespace std;

int num;
int ls[MAX];
int numbers[MAX];
int main() {
  scanf("%d",&num);
  for(int i = 0; i < num; i++) {
    scanf("%d",&(ls[i]));
    numbers[i] = ls[i];
  }
  sort(numbers,numbers+num,less<int>());
  int pos = 0;
  int pos_num = 0;
  string act = "";
  stack<int> s;
  while(pos<num) {
    while(s.empty() || s.top() != ls[pos]) {
      if(pos_num == num) {
        printf("NONE\n");
        return 0;
      }
      s.push(numbers[pos_num]);
      pos_num++;
      act += "+\n";
    }
    s.pop();
    act += "-\n";
    pos++;
  }
  cout << act;
  return 0;
}
